### ISSUE TITLE
HVAC controller integration ID

### ISSUE THREAD

**Issue by johannesparty** [senorshaun/homebridge-radiora2#35]
Hi,

(help request)

I have seeTemp controls, RadioRA2 HVAC controller, and some wireless temp sensors.
I cannot find the seeTemp or HVAC controller integration id when I produce an integration report.
The temp sensor does show up and looks like it might work with the plugin.

I have attached a copy of my integration report 
(there should be 3 seeTemp, 3 HVAC controller, and 3 Temp Sensors ... but can only find the temp sensors).

Thanks!

[lutron_integraton_20231029.csv](https://github.com/senorshaun/homebridge-radiora2/files/13200327/lutron_integraton_20231029.csv)

**Comment by johannesparty** [comment]
New info: "discover all devices" does not work (hangs forever ... waited several hours ... no obvious progress).  I can send logs if you want, let me know.
 
New info: was able to "guess" integration numbers and seems to work (73, 75, 112 in supplied integration CSV above).

Minor bug: doesn't show correct set point until set point changes.

**Comment by senorshaun** [comment]
So it sounds the 'Zone' that you have the hvac controller in is the integration id?

For the discover devices, can you go to "https://[repeaterIPaddress]/DbXmlInfo.xml" and send me that data?

**Comment by johannesparty** [comment]
From Browser, looks odd (see picture below).
But when I load with curl, it seems fine (attached; bigger than I expected).

Maybe my naming for the HVAC controller was a poor choice, but don't really
want to change it now since everything is working.
The integration seems to be working well (other than not showing current
set point until it changes; minor bug but still a bug).

-Johannes.


[image: image.png]

Johannes Schmidt
***@***.***
408-621-0091

**Comment by senorshaun** [comment]
It looks like your attachments didn't attach through email

**Comment by johannesparty** [comment]
"Sent" from gmail web interface shows it was attached, but resending now using Apple Mail client to see if that helps.

￼


Johannes Schmidt
408-621-0091

**Comment by senorshaun** [comment]
no. just go to the GitHub page and attach it. or send to me at shaun @ therowan . house

**Comment by johannesparty** [comment]
Here is the file.  However, the ID's seem to work, so this isn't super urgent.  Feel free to close if you want.

[lutron_integraton_20231029.csv](https://github.com/senorshaun/homebridge-radiora2/files/13231393/lutron_integraton_20231029.csv)

**Comment by senorshaun** [comment]
that looks like the integration report again. Can you please copy and paste the xml from the webpage? or curl it and save to a file?

**Comment by johannesparty** [comment]
Sorry, here it is.

[DbXmlInfo.xml.tar.gz](https://github.com/senorshaun/homebridge-radiora2/files/13231790/DbXmlInfo.xml.tar.gz)

**Comment by senorshaun** [comment]
I would like to get this working. You have more HVAC stuff than I ever messed with, so the logs from when the Discover all devices stalls out would be great. It should only take like 10 seconds or so to complete, you don't need to wait hours

**Comment by senorshaun** [comment]
Also, can you help me with how your equipment is set up? From what I understand the seeTemp is just a way to view the current temp of an HVAC controller and change the set points. If that is correct, we wouldn't have that in HomeKit because we already have the HVAC controller. Additionally, the HVAC controller doesn't actually have a temperature sensor itself right? you have to pair it with the wireless temp sensor for it to work properly. So for all the equipment, you should only get 3 thermostats that show current temp in HomeKit right? Technically, your HVAC controller could control temp for multiple rooms, so I think if you modify the config I have the option to create multiple temp sensors, all displaying the same temp, so you can put one in each 'room' in homekit that the HVAC controller affects. Personally I wouldn't do that, I would just put the thermostat in the 'room' that cooresponds to where you actually have the wireless temp sensor, not the controller (which is likely in a crawlspace or attic). 

What are the names of your HVAC controllers and the corresponding wireless temp sensor? I'm trying to figure out how to automatically discover the relationship between those, and right now I'm not seeing anything obvious

**Comment by johannesparty** [comment]
Setup: radiora2 inclusive because I needed more than 100 devices (probably doesn't matter)
- Three HVAC controllers all in "Big Wiring Closet" (LR-HVAC-1-WH) configured for heat only. 
- Each in a different zone ("Upstairs", "Living Room", "Master Bed").  not the best names as the living room and master bed zones each cover multiple rooms, just like "upstairs" covers multiple rooms.  In "program" section setup you list rooms each of the HVAC controller affects.
- Three seeTemp keypads (you probably don't care)
- Three Temperature Sensors (LRF2-TWRB-SW) in rooms ("Living Room", "Bed 1", "Bed 3").
- Programming section for each HVAC sensor associates with one temp sensor each (multiple are possible and I think the HVAC controller just averages).

So ... I'm really not sure if there is a reason for me to give you the temp sensors rather than just getting the value from the HVAC controller.  
- Maybe if you have multiple temp sensors for one HVAC and want to see the individual values.  
- Also when HVAC is set to "off" instead of "heat", the seeTemp and Lutron's own application doesn't show the temp.  So I usually leave HVAC on to HEAT, but set a super low temperature so it never activates.  Maybe linking to a temp controller directly would let me see temp even when HVAC is set to "off".

I can do some experimentation to remove a temp sensor from the config and see what things look like.  I'm _guessing_ for most people, maybe just link the HVAC and not Temp Sensor, but really need to confirm.

Let me know what I can do to help you debug the discovery part, in case the Xml wasn't enough.  Also apologies for messing up getting that to you so often!  Thanks for writing this plugin!
